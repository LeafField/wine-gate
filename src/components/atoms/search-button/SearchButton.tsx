import React from "react";

const SearchButton = () => {
  return (
    <button
      type="submit"
      className="border-2 border-gray px-3 py-1 transition-[border-color] duration-300 hover:border-blue focus:border-blue focus:outline-none"
    >
      検索
    </button>
  );
};

export default SearchButton;
