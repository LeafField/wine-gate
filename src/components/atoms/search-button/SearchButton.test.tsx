import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import SearchButton from "./SearchButton";

describe("SearchButtonの単体テスト", () => {
  test("コンポーネントが正しくレンダリングされているか", () => {
    render(<SearchButton />);
    const button = screen.getByRole("button", { name: /検索/i });
    expect(button).toBeInTheDocument();
  });

  test("ボタンがクリック可能であること", () => {
    render(<SearchButton />);
    const button = screen.getByRole("button", { name: /検索/i });
    userEvent.click(button);
    expect(button).toBeEnabled();
  });

  test("スナップショットテスト", () => {
    const { container } = render(<SearchButton />);
    expect(container).toMatchSnapshot();
  });
});
