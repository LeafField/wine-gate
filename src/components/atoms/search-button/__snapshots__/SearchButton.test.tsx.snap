// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButtonの単体テスト スナップショットテスト 1`] = `
<div>
  <button
    class="border-2 border-gray px-3 py-1 transition-[border-color] duration-300 hover:border-blue focus:border-blue focus:outline-none"
    type="submit"
  >
    検索
  </button>
</div>
`;
